execute/table.go

```diff
@@ -953,6 +1018,45 @@ func (b ColListTableBuilder) RawTable() *ColListTable {
 	return b.table
 }
 
+// SliceColumns iterates over each column of b and re-slices them to the range
+// [start:stop].
+func (b ColListTableBuilder) SliceColumns(start, stop int) error {
+	if start < 0 || start > stop {
+		return fmt.Errorf("invalid start/stop parameters: %d/%d", start, stop)
+	}
+
+	if stop < start || stop > b.table.nrows {
+		return fmt.Errorf("invalid start/stop parameters: %d/%d", start, stop)
+	}
+
+	for i, col := range b.table.cols {
+		switch col.Meta().Type {
+		case flux.TBool:
+			col := b.table.cols[i].(*boolColumn)
+			col.data = b.alloc.SliceBools(col.data, start, stop)
```

Comment:
I see why you are using the allocator to track that the memory is not being used anymore. To reduce complexity and avoid possible accounting errors I think we should not use the allocator to manage slices.

But rather when we call Free inside the `Clear` call for the various table implementations we should use the capacity of the array not the length. This should keep accounting simple and allow for slicing the array in the meantime. 

Also to confirm my understanding that slicing cannot cause new allocations here is the relevant bit in the Go spec https://golang.org/ref/spec#Slice_types.